master/internal/kubernetes/pod_test.go

```diff
@@ -0,0 +1,795 @@
+	message, err := podMap["task"].Pop()
+	if err != nil {
+		t.Errorf("Unable to pop message from task receiver queue")
+	}
+
+	containerMsg, ok := message.(sproto.ContainerLog) // sproto.StopPod
+	if !ok {
+		t.Errorf("expected sproto.ContainerLog but received %s", reflect.TypeOf(message))
+	}
+	assert.Equal(t, *containerMsg.AuxMessage, correctMsg)
+	err = os.Remove("k8_init_container_entrypoint.sh")
+	if err != nil {
+		t.Logf("Failed to remove entrypoint")
+	}
+}
+
+func TestReceivePodStatusUpdateTerminated(t *testing.T) {
+	setupEntrypoint(t)
+
+	// Pod deleting, but in pending state
+	t.Logf("Testing PodPending status")
+	system, newPod, ref, podMap, _ := createPodWithMockQueue()
+	podMap["task"].Purge()
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+
+	typeMeta := metaV1.TypeMeta{Kind: "rest test"}
+	objectMeta := metaV1.ObjectMeta{
+		Name:              "test meta",
+		DeletionTimestamp: &metaV1.Time{Time: time.Now()},
+	}
+	pod := k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     k8sV1.PodStatus{Phase: k8sV1.PodPending},
+	}
+
+	statusUpdate := podStatusUpdate{updatedPod: &pod}
+
+	checkReceiveTermination(t, statusUpdate, system, ref, newPod, podMap)
+
+	// Pod failed
+	t.Logf("Testing PodFailed status")
+	system, newPod, ref, podMap, _ = createPodWithMockQueue()
+	podMap["task"].Purge()
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+	pod = k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     k8sV1.PodStatus{Phase: k8sV1.PodFailed},
+	}
+	statusUpdate = podStatusUpdate{updatedPod: &pod}
+
+	checkReceiveTermination(t, statusUpdate, system, ref, newPod, podMap)
+
+	// Pod succeeded
+	system, newPod, ref, podMap, _ = createPodWithMockQueue()
+	podMap["task"].Purge()
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+	pod = k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     k8sV1.PodStatus{Phase: k8sV1.PodSucceeded},
+	}
+	statusUpdate = podStatusUpdate{updatedPod: &pod}
+
+	checkReceiveTermination(t, statusUpdate, system, ref, newPod, podMap)
+
+	system.Ask(ref, statusUpdate)
+	time.Sleep(time.Second)
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+
+	err := os.Remove("k8_init_container_entrypoint.sh")
+	if err != nil {
+		t.Logf("Failed to remove entrypoint")
+	}
+}
+
+func TestMultipleContainerTerminate(t *testing.T) {
+	// Status update test involving two containers
+	setupEntrypoint(t)
+
+	// Pod running with > 1 container, and one terminated
+	t.Logf("Testing two pods with one in terminated state")
+	system, newPod, ref, podMap, _ := createPodWithMockQueue()
+	containerStatuses := []k8sV1.ContainerStatus{
+		{
+			Name: "test-pod-1",
+			State: k8sV1.ContainerState{
+				Running: &k8sV1.ContainerStateRunning{},
+			},
+		},
+		{
+			Name: "test-pod-2",
+			State: k8sV1.ContainerState{
+				Terminated: &k8sV1.ContainerStateTerminated{},
+			},
+		},
+	}
+	newPod.containerNames = map[string]bool{"test-pod-1": false, "test-pod-2": false}
```

Comment:
should update one these to be `determined-container` since that is what the pod actor uses to decide the state of the pod